tests_integration/test_main.py

```diff
@@ -542,3 +545,73 @@ def test_protocols_udp_restricted_port_values_start_must_be_lte_end(
 
         stderr = ex.value.stderr.decode()
         assert "Start port value must be less or equal to end port value" in stderr
+
+
+@pytest.mark.integration()
+def test_resource_access_flows(kopf_settings, unique_resource_name):
+    assert "TWINGATE_PRINCIPAL_ID" in os.environ
+    principal_id = os.environ["TWINGATE_PRINCIPAL_ID"]
+
+    RESOURCE_OBJ = f"""
+        apiVersion: twingate.com/v1beta
+        kind: TwingateResource
+        metadata:
+          name: {unique_resource_name}
+        spec:
+          name: My K8S Resource
+          address: my.default.cluster.local
+    """
+
+
+    OBJ = f"""
+        apiVersion: twingate.com/v1beta
+        kind: TwingateResourceAccess
+        metadata:
+          name: {unique_resource_name}
+        spec:
+          resourceRef:
+            name: {unique_resource_name}
+          principalId: {principal_id}
+    """
+
+    # fmt: off
+    with KopfRunner(kopf_runner_args, settings=kopf_settings) as runner:
+        kubectl_create(RESOURCE_OBJ)
+        time.sleep(10)  # give it some time to react
+
+        kubectl_create(OBJ)
+        time.sleep(5)  # give it some time to react
+
+        created_object = json.loads(kubectl(f"get tgra/{unique_resource_name} -o json").stdout)
+
+        delete_command = kubectl(f"delete tgra/{unique_resource_name}")
+
+        time.sleep(1)  # give it some time to react
+
+        resource_delete_command = kubectl(f"delete tgr/{unique_resource_name}")
+
+    # fmt: on
+
+    # Ensure that the operator did not die on start, or during the operation.
+    assert runner.exception is None
+    assert runner.exit_code == 0
+
+    logs = load_stdout(runner.stdout)
+    for log in logs:
```

Comment:
This is a debug leftover - should delete it from `test_resource_flows` too

Reply:
recommended change applied